core/src/main/cpp/jni/projection_mesh/rectangular_projection_mesh_jni.cpp

```diff
@@ -0,0 +1,63 @@
+#include <jni.h>
+
+#include "gdn/gast_node.h"
+#include "gdn/projection_mesh/rectangular_projection_mesh.h"
+#include "utils.h"
+
+// Current class and package names assumed for the Java side.
+#undef JNI_PACKAGE_NAME
+#define JNI_PACKAGE_NAME org_godotengine_plugin_gast_projectionmesh
+
+#undef JNI_CLASS_NAME
+#define JNI_CLASS_NAME RectangularProjectionMesh
+
+namespace {
+using namespace gast;
+using namespace godot;
+
+inline RectangularProjectionMesh *from_mesh_pointer(jlong rectangular_projection_mesh_pointer) {
+    return reinterpret_cast<RectangularProjectionMesh *>(rectangular_projection_mesh_pointer);
+}
+
+inline GastNode *from_node_pointer(jlong gast_node_pointer) {
+    return reinterpret_cast<GastNode *>(gast_node_pointer);
+}
+
+}  // namespace
+
+extern "C" {
+
+JNIEXPORT void JNICALL
+JNI_METHOD(nativeSetMeshSize)(JNIEnv *, jobject, jlong mesh_pointer, jlong node_pointer, jfloat width, jfloat height) {
+    RectangularProjectionMesh *mesh = from_mesh_pointer(mesh_pointer);
+    GastNode *gast_node = from_node_pointer(node_pointer);
+    ERR_FAIL_NULL(mesh);
+    ERR_FAIL_NULL(gast_node);
+    mesh->set_mesh_size(Vector2(width, height));
+    gast_node->setup_projection_mesh();
```

Comment:
Here and below, this call should be done in the `ProjectionMesh` methods, for example in `set_mesh_size` for this scenario.
The issue with doing it here is that this jni path is not invoked when the calls are made from the Godot runtime side leading to an invalid state.

If the issue has to do with how to access the `GastNode` object, you can add a setter for that purpose to the `ProjectionMesh` class and invoke it within `GastNode::set_projection_mesh(...)`.

You will also have to take care to unset the `GastNode` object from any projection mesh that's being replaced. Also don't forget to update the default projection mesh that's set in `GastNode`'s constructor. It may be safer to just call `set_projection_mesh` from within the constructor to ensure that the same logic is applied.

Reply:
Used listeners as discussed offline to avoid circular dependency.